microsoft/testsuites/core/cpu.py

```
@@ -129,3 +132,63 @@ def l3_cache_check(self, node: Node, log: Logger) -> None:
+            interrupts by checking the /proc/interrupts file.
+
+            Steps:
+            1. Looks for the Hyper-v timer property of each CPU under /proc/interrupts
+            2. Verifies if atleast one CPU has more than 0 interrupts processed.
+
+            Note: There are 3 types of Hyper-v interrupts : Hypervisor callback
+            interrupts, Hyper-V reenlightenment interrupts, and Hyper-V stimer0
+            interrupts. We also do not need to have each CPU handle vmbus message
+            or event in multi CPUs VM.
+            """,
+        priority=2,
+    )
+    def verify_vmbus_interrupts(self, node: Node, log: Logger) -> None:
+        found_hyperv_interrupt = False
+        cpu_count = node.tools[Lscpu].get_core_count()
+        is_cpu_handling_interrupt = [False] * cpu_count
+        log.debug(f"{cpu_count} CPU cores detected...")
+
+        interrupts = node.tools[Cat].get_interrupt_data()
+        for interrupt in interrupts:
+            is_hyperv_interrupt = any(
+                [(substr in interrupt.metadata) for substr in hyperv_interrupt_substr]
+            )
+            found_hyperv_interrupt = found_hyperv_interrupt | is_hyperv_interrupt
+            if is_hyperv_interrupt:
+                log.debug(f"Hyper-V interrupt : {interrupt}")
+                assert_that(
+                    len(interrupt.interrupt_count),
+                    "Interrupt count should be present for each cpu.",
+                ).is_equal_to(cpu_count)
+                is_cpu_handling_interrupt = [
+                    ((interrupt.interrupt_count[i] > 0) | is_cpu_handling_interrupt[i])
+                    for i in range(len(interrupt.interrupt_count))
+                ]
+
+        log.debug(f"CPU interrupt handling : {is_cpu_handling_interrupt}")
+
+        # It is not mandatory to have the Hyper-V interrupts present under
+        # `/proc/interrupts`. Skip test execution if these are not showing up
+        if not found_hyperv_interrupt:
```

The test logic looks strange to me. If some of interrupts are not found, it fails. But if none of them found, it's skipped. Is there some way to check it more strictly? It looks need to differentiate the real skipped from all of interrupts are not queried unexpectedly?

---

We are only checking hyper-v interrupts here. Based on the comments in the v2 test file, hyper-v interrupts might not be recorded for every distro, in which case we skip the test. However, if the interrupts are recorded, they should be handled by atleast one CPU.

---

Based on offline discussion, we will set the state to failed.

---

I forgot what we discussed. If the state should be "failed", should it raise LisaException instead of SkippedException?

---

iirc we wanted to mark the test as failed if some distro doesn't have any hyper-v interrupt present and understand which distros show such behavior. I will change skip to failed.